tests/integration/test_join.py

```diff
@@ -184,3 +184,358 @@ def test_join_literal(c):
+            ],
+            [
+                1,
+                3,
+                pd.NA,
+                334,
+                "c4",
+                np.nan,
+                pd.Timestamp("2003-04-01"),
+                2,
+                5,
+                pd.NA,
+                112,
+                "a3",
+                np.nan,
+                pd.Timestamp("2001-03-01"),
+            ],
+            [
+                1,
+                4,
+                35,
+                335,
+                "c5",
+                np.nan,
+                pd.Timestamp("2003-05-01"),
+                2,
+                5,
+                pd.NA,
+                112,
+                "a3",
+                np.nan,
+                pd.Timestamp("2001-03-01"),
+            ],
+            [
+                1,
+                4,
+                35,
+                335,
+                "c5",
+                np.nan,
+                pd.Timestamp("2003-05-01"),
+                4,
+                6,
+                13,
+                113,
+                "a4",
+                np.nan,
+                pd.Timestamp("2001-04-01"),
+            ],
+            [
+                2,
+                1,
+                36,
+                336,
+                "c6",
+                np.nan,
+                pd.Timestamp("2003-06-01"),
+                pd.NA,
+                pd.NA,
+                pd.NA,
+                pd.NA,
+                np.nan,
+                np.nan,
+                pd.NaT,
+            ],
+            [
+                2,
+                3,
+                37,
+                337,
+                "c7",
+                np.nan,
+                pd.NaT,
+                pd.NA,
+                pd.NA,
+                pd.NA,
+                pd.NA,
+                np.nan,
+                np.nan,
+                pd.NaT,
+            ],
+            [3, 2, 38, 338, "c8", 3.8, pd.NaT, 1, 2, 14, 114, "a5", np.nan, pd.NaT],
+            [3, 2, 39, 339, "c9", 3.9, pd.NaT, 1, 2, 14, 114, "a5", np.nan, pd.NaT],
+            [3, 2, 38, 338, "c8", 3.8, pd.NaT, 2, 3, 15, 115, "a6", 1.6, pd.NaT],
+            [3, 2, 39, 339, "c9", 3.9, pd.NaT, 2, 3, 15, 115, "a6", 1.6, pd.NaT],
+        ],
+        columns=[
+            "ids",
+            "dates",
+            "pn_nullint",
+            "pn_int",
+            "pn_str",
+            "pn_float",
+            "pn_date",
+            "startdate",
+            "enddate",
+            "lk_nullint",
+            "lk_int",
+            "lk_str",
+            "lk_float",
+            "lk_date",
+        ],
+    )
+    change_types = {
+        "pn_nullint": "Int32",
+        "lk_nullint": "Int32",
+        "startdate": "Int64",
+        "enddate": "Int64",
+        "lk_int": "Int64",
+        "pn_str": "string",
+        "lk_str": "string",
+    }
+    for k, v in change_types.items():
+        dfcorrpn[k] = dfcorrpn[k].astype(v)
+
+    # Left Join
+    querypnl = """
```

Comment:
I have tested this Query using PostgresSQL, it somewhat matches with the correct dataset specified here except for some `dtype` mismatch,  I have one suggestion can Please you add this testing on `test_postgres.py`  as well, where you can make use of `assert_query_gives_same_result` to compare your result directly with Postgres result. let me know if you need any help.